src/OpenLoco/Math/Vector.hpp

```diff
@@ -0,0 +1,293 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include <string>
+
+namespace OpenLoco::Math::Vector
+{
+#pragma pack(push, 1)
+    template<typename T, T TResolution>
+    struct TVector2
+    {
+        static constexpr auto Resolution = TResolution;
+
+        T x = 0;
+        T y = 0;
+
+        constexpr TVector2() = default;
+        constexpr TVector2(T _x, T _y)
+            : x(_x)
+            , y(_y)
+        {
+        }
+
+        template<T TResolution2>
+        constexpr TVector2(const TVector2<T, TResolution2>& other)
+            : TVector2(other.x, other.y)
+        {
+            if constexpr (TResolution < TVector2<T, TResolution2>::Resolution)
+            {
+                x *= TVector2<T, TResolution2>::Resolution;
+                y *= TVector2<T, TResolution2>::Resolution;
+            }
+            if constexpr (TResolution > TVector2<T, TResolution2>::Resolution)
+            {
+                x /= TResolution;
+                y /= TResolution;
+            }
+        }
+
+        constexpr bool operator==(const TVector2& rhs) const
+        {
+            return x == rhs.x && y == rhs.y;
+        }
+
+        constexpr bool operator!=(const TVector2& rhs) const
+        {
+            return !(*this == rhs);
+        }
+
+        constexpr TVector2& operator+=(const TVector2& rhs)
+        {
+            x += rhs.x;
+            y += rhs.y;
+            return *this;
+        }
+
+        constexpr TVector2& operator-=(const TVector2& rhs)
+        {
+            x -= rhs.x;
+            y -= rhs.y;
+            return *this;
+        }
+
+        constexpr TVector2& operator*=(const T rhs)
+        {
+            x *= rhs;
+            y *= rhs;
+            return *this;
+        }
+
+        constexpr TVector2& operator/=(const T rhs)
+        {
+            x /= rhs;
+            y /= rhs;
+            return *this;
+        }
+
+        constexpr bool operator>=(const TVector2& rhs) const
```

Comment:
I still don't see the point of these comparisons but I'm not overly fussed. I guess it might be useful for some ordering container.

Reply:
Removed the unused operators